core/src/main/java/org/elasticsearch/common/rounding/Rounding.java

```diff
@@ -128,20 +128,40 @@ public byte id() {
         @Override
         public long round(long utcMillis) {
             long rounded = field.roundFloor(utcMillis);
-            if (timeZone.isFixed() == false && timeZone.getOffset(utcMillis) != timeZone.getOffset(rounded)) {
-                // in this case, we crossed a time zone transition. In some edge
-                // cases this will
-                // result in a value that is not a rounded value itself. We need
-                // to round again
-                // to make sure. This will have no affect in cases where
-                // 'rounded' was already a proper
-                // rounded value
-                rounded = field.roundFloor(rounded);
+            if (timeZone.isFixed() == false) {
+                if (timeZone.getOffset(utcMillis) != timeZone.getOffset(rounded)) {
+                    // in this case, we crossed a time zone transition. In some
+                    // edge cases this will result in a value that is not a rounded
+                    // value itself. We need to round again to make sure. This will have no effect in
+                    // cases where 'rounded' was already a proper rounded value
+                    rounded = field.roundFloor(rounded);
+                } else {
+                    if (onDSTOverlap(rounded)) {
```

Comment:
I think generally "skipping" (or merging, like here) intervals that are less than the field duration is this is not always the right thing. Imagine - for example - a day that due to DST shift has only 23h (typically on DST start). We don't want to join that with the previous or the following day. I can look into this once again but its very likely some of our existing tests blow up when doing this.